File: 88.合并两个有序数组.py
#
# @lc app=leetcode.cn id=88 lang=python
#
# [88] 合并两个有序数组
#

# 从后往前做归并就行
# @lc code=start
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        cur = len(nums1) - 1
        l1, l2 = m - 1, n - 1
        while l1 >= 0 and l2 >= 0:
            if nums1[l1] > nums2[l2]:
                nums1[cur] = nums1[l1]
                l1 -= 1
            else:
                nums1[cur] = nums2[l2]
                l2 -= 1
            cur -= 1
        while l2 >= 0:
            nums1[cur] = nums2[l2]
            cur -= 1
            l2 -= 1

# @lc code=end

File: 88.合并两个有序数组.cpp
/*
 * @lc app=leetcode.cn id=88 lang=cpp
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if (n == 0) {
            return;
        }
        int idx = nums1.size() - 1;
        int l1 = m - 1;
        int l2 = n - 1;
        while (l1 >= 0 && l2 >= 0) {
            if (nums1[l1] < nums2[l2]) {
                nums1[idx] = nums2[l2];
                l2--;
            } else {
                nums1[idx] = nums1[l1];
                l1--;
            }
            idx--;
        }
        while (l2 >= 0) {
            nums1[idx--] = nums2[l2--]; 
        }
    }
};
// @lc code=end

